### PR TITLE
Add drag and drop

### LINKED ISSUE #67: Add drag and drop for PosterCards into the sidebar
The user should be able to add a movie/show to a list by dragging a poster card into the list in the sidebar (desktop only, not sure if it's even possible to drag stuff on mobile browsers).

### PR DESCRIPTION
**Pull request by willwull**
* PosterCards are now draggable using the library `react-dnd`
  * I had to rewrite PosterCard into a class component since `react-dnd` needs to use `refs` which are only available in class components
* They can be dropped into the list links in the sidebar
  * Since the logic for adding to lists in firebase isn't implemented yet, I just added a placeholder so that the action fails randomly
* Added a text at the bottom of the sidebar explaining that you can drag and drop
* So called "toast notifications" are shown when adding, implemented with the library `react-toastify`.

fixes #67

### CODE REVIEW

File: src/components/PosterCardDropTarget.js [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+      </div>
+    ));
+  }
+}
+
+const listTarget = {
+  // is called when a PosterCard is dropped on this component
+  drop(props, monitor) {
+    const item = monitor.getItem();
+    const success = Math.random() > 0.3;
```
Comment (iRobsc) [inline src/components/PosterCardDropTarget.js]: We shouldn't merge test cases to master

Comment (willwull) [reply src/components/PosterCardDropTarget.js]: What should we have instead?

Comment (iRobsc) [reply src/components/PosterCardDropTarget.js]: It should always show that it's successful before we added real states where there could be errors.

Comment (willwull) [reply src/components/PosterCardDropTarget.js]: Okay, I added this mostly to show that `errorToast()` exists, but I'll fix this.

**Pull request merged**